frontend/packages/dev-console/src/components/pipelines/list-page/PipelineRow.tsx

```diff
@@ -10,10 +10,30 @@ import { getPipelineKebabActions } from '../../../utils/pipeline-actions';
 import LinkedPipelineRunTaskStatus from '../../pipelineruns/status/LinkedPipelineRunTaskStatus';
 import { ResourceKebabWithUserLabel } from '../../pipelineruns/triggered-by';
 import { tableColumnClasses } from './pipeline-table';
+import { usePipelineTriggerTemplateNames } from '../utils/triggers';
 
 const pipelineReference = referenceForModel(PipelineModel);
 const pipelinerunReference = referenceForModel(PipelineRunModel);
 
+type PipelineRowKebabActionsProps = {
+  pipeline: Pipeline;
+};
+
+const PipelineRowKebabActions: React.FC<PipelineRowKebabActionsProps> = ({ pipeline }) => {
```

Comment:
This needs to be in a new file. Also as discussed on slack, we need to lazy do this as we cannot be fetching always for every row... it's too much.

Reply:
@vikram-raj We discussed this on slack. For posterity, I'll look into this more this weekend. There is something off about this and I will figure out what it is or retract my request 😛

Reply:
Looks like there is caching in play here that prevents too much duplicate action. We still need to convert this component into it's own file, but let's run with this design for now. It doesn't seem to have too much problems. At least none more than the current Pipeline (which fetches all Pipeline Runs and gets the latest).

Reply:
Moved the ```PipelineRowKebabActions``` to separate files.